binscripts/rvm-installer

```diff
@@ -257,9 +259,11 @@ fetch_version()
       return 0
     fi
   done
+  notice "Exhausted all sources trying to fetch version of RVM!"
```

Comment:
This should rather use `fail_with_code`, right?

Reply:
That's a fair Q as well, and relates directly to https://github.com/rvm/rvm/pull/4728#discussion_r302506596. I can definitely simplify some logic by just using `fail_with_code` here and be over & done with it.  I was just hesitant to add an implicit `exit` to a function where it hadn't been there yet before.

But there are Pros for either approach.  What's your pref?  I am happy to lean either way.

Reply:
@mpapis what you think?

Reply:
`notice` and `log` feels quite similar just by looking at the name, so I would not bring `notice` on board

Reply:
I was thinking it would be helpful to log problems to STDERR instead of STDOUT.  The thought being to use the same output stream as `fail`, but without halting.

Reply:
I would let @mpapis decide on this.

But if we keep it, we should rather call the function `error` not `notice`, don't you think?

Reply:
Good call.  Though only trouble with `error` is that my change on line 285 becomes especially awkward, when in fact in that case we don't yet _know_ whether the request is going to fail or succeed:

    error "Fetching ${_url}"